zproject/backends.py

```diff
@@ -351,10 +352,15 @@ def authenticate(self, request: HttpRequest, *,
         user_profile = common_get_active_user(username, realm, return_data=return_data)
         if user_profile is None:
             return None
-        if user_profile.check_password(password):
-            if hasattr(request, 'session'):
-                request.session['needs_to_change_password'] = not check_password_strength(password)
-                return user_profile
+        try:
+            if user_profile.check_password(password):
+                if hasattr(request, 'session'):
```

Comment:
hmm I might be confused at this point, but didn't we end up deciding in a chat to get rid of the `hasattr` check and just assume `request` has a session?

Reply:
Oops, sorry. I thought I haven't committed that change. Changed it.